projects/pastanaga-angular/src/lib/controls/textfield/base-text-field.ts

```diff
@@ -238,4 +245,10 @@ export class BaseTextField extends BaseControl implements AfterContentInit, OnIn
         this.cdr.markForCheck();
         this.debouncer.next(value);
     }
+
+    protected isUntouched() {
+        return !!this.element && !!this.element.nativeElement
+            && !!this.element.nativeElement.classList
+            && this.element.nativeElement.classList.contains('ng-untouched');
```

Comment:
is this class is there in production mode as well?

Reply:
it's a class automatically added by angular on the containers